Add decoding commands to dapf (#688)
cloudflare/daphne: crates/dapf/README.md

# dapf (DAP functions)

This binary provides various functions that can be called to interact with DAP
deployments.

> It several subcommands and their function can be consulted with the built in
> help pages the CLI provides.

Reviewer: ```suggestion
Its several subcommands and their functionality can be consulted with the built in
help pages the CLI provides.
```

## Examples

Here are some examples of common functionality of dapf. To avoid installing the
tool replace all instances of `dapf` with `cargo run --bin dapf`.

### Generate an hpke config

With the default algorithm
```
dapf hpke generate
```

With a specific algorithm
```
dapf hpke generate x25519_hkdf_sha256
```

This will output the config twice
```
{
  "config": {
    "id": 131,
    "kem_id": "x25519_hkdf_sha256",
    "kdf_id": "hkdf_sha256",
    "aead_id": "aes128_gcm",
    "public_key": "3ccc3ada22dc0b3367d031ea4b02876a77a59ef32b5a1c06109552a51a739a13"
  },

>   "private_key": "PRIVATE_KEY"

Reviewer: nit: might as well leave the actual generated key in, since this is just a demonstration

}

> hpke config prio encoded and base64 encoded: gwAgAAEAAQAgPMw62iLcCzNn0DHqSwKHanelnvMrWhwGEJVSpRpzmhM

Reviewer: "prio encoded"? Here and below.

```

First as JSON and second as a prio encoded + base64 encoded string which can be
used when issuing requests to `internal/test/add_task`

### Decoding responses from aggregators

> 
> #### Decoding aggregate shares
> Once you've collected from the leader you may want to see what the result of the

Reviewer: ```suggestion
#### Decoding aggregate shares

Once you've collected from the leader you may want to see what the result of the
```
